src/analysis/convex/function.lean

```diff
@@ -104,6 +104,15 @@ lemma strict_concave_on.subset {t : set E} (hf : strict_concave_on 𝕜 t f) (hs
   strict_concave_on 𝕜 s f :=
 ⟨hs, λ x y hx hy, hf.2 (hst hx) (hst hy)⟩
 
+section composition
+variables [has_smul 𝕜 Q] [ordered_add_comm_monoid Q] (t : set β) (g : β → Q)
+
+lemma convex_on.compose (hf : convex_on 𝕜 s f) (hg : convex_on 𝕜 t g) (hg' : monotone g)
```

Comment:
Probably, you only need `monotone_on g t`.

Reply:
I don't think I can use ```monotone_on g t``` because ```a \smul f x + b \smul f y``` may not be in ```t```.

Reply:
What you want is `monotone_on (f '' s) g`.